src/transforms/transform-deployment.ts

```diff
@@ -155,6 +157,34 @@ export function mapEnvironment(environment: string): string {
 	return jiraEnv;
 }
 
+// Maps commit summaries to an array of a single association containing the commit keys (commit hash and repository id)
+const mapCommitSummariesToAssociationArray = (
+	repositoryId: string,
+	commitSummaries?: CommitSummary[]
+): JiraAssociation[] | undefined => {
+
+	let associationArray;
+
+	if (commitSummaries && commitSummaries.length) {
```

Comment:
better way of doing this would be to add the condition check to return undefined at the top instead of wrapping most of the function in an conditional:

`if(commitSummaries?.length) { return undefined; }` at the top will do, and the rest can just continue as in.

There's also the question, should the function return undefined or just an empty array?  Would the devinfo api error on an empty array?

Reply:
I did a quick test, and it appears that receiving an empty array causes issues, so it's safer to not send it at all - ie. set to undefined, rather than []

Reply:
yeah, figured as much